File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-0.py
# Definition for singly-linked list.
class ListNode:
     def __init__(self, val=0, next=None):
         self.val = val
         self.next = next

class Solution:    
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        nums = []
        for li in lists:
            while li != None:
                nums.append(li.val)
                li = li.next
        nums.sort()
        if len(nums) == 0:
            return None
        last = ListNode(nums[-1])
        nums = nums[::-1]
        for i in range(1, len(nums)):
            cur = ListNode(nums[i], last)
            last = cur
        return last

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        int cnt[2001] = {0};
        vector<int> keys;
        for(int ni: nums) {
            if(cnt[ni + 1000] == 0) keys.push_back(ni);
            cnt[ni + 1000] += 1;
        }
        sort(keys.begin(), keys.end(), [&](int a1, int a2){
            return (cnt[a1 + 1000] > cnt[a2 + 1000]);
        });
        vector<int> ans;
        for(int i = 0; i < k; ++i) ans.push_back(keys[i]);
        return ans;
    }
};
